packages/kbn-esql-utils/src/utils/append_to_query.ts

```diff
@@ -5,9 +5,73 @@
  * in compliance with, at your election, the Elastic License 2.0 or the Server
  * Side Public License, v 1.
  */
+import { getAstAndSyntaxErrors } from '@kbn/esql-ast';
 
 // Append in a new line the appended text to take care of the case where the user adds a comment at the end of the query
 // in these cases a base query such as "from index // comment" will result in errors or wrong data if we don't append in a new line
 export function appendToESQLQuery(baseESQLQuery: string, appendedText: string): string {
   return `${baseESQLQuery}\n${appendedText}`;
 }
+
+export function appendWhereClauseToESQLQuery(
+  baseESQLQuery: string,
+  field: string,
+  value: unknown,
+  operation: '+' | '-',
+  fieldType?: string
+): string {
+  let operator = operation === '+' ? '==' : '!=';
+  let filterValue = typeof value === 'string' ? `"${value.replace(/"/g, '\\"')}"` : value;
+  // Adding the backticks here are they are needed for special char fields
+  let fieldName = `\`${field}\``;
+
+  // casting to string
+  // there are some field types such as the ip that need
+  // to cast in string first otherwise ES will fail
+  if (fieldType !== 'string' && fieldType !== 'number' && fieldType !== 'boolean') {
+    fieldName = `${fieldName}::string`;
+  }
```

Comment:
Can we limit this functionality to field types we are sure about (string, number, ip etc) and hide filter buttons for all others? 

<img width="1579" alt="Screenshot 2024-04-30 at 14 38 04" src="https://github.com/elastic/kibana/assets/1415710/e46d26ee-2e60-4a2a-b6f3-306f1480c84b">